### ISSUE TITLE
Core dumped while reading graph with NetworkitBinaryReader if remove nodes before write graph

### ISSUE THREAD

**Issue by Jerevia** [networkit/networkit#963]
For example,
```python
import networkit as nk

nkG = nk.Graph(weighted=True)

nkG.addNodes(100)
nkG.addEdge(0, 1)
nkG.addEdge(2, 3)
nkG.removeNode(10)

nk.writeGraph(nkG, 'graph.bin', nk.Format.NetworkitBinary)
nk.readGraph('graph.bin', nk.Format.NetworkitBinary)
```
it prints
```
Segmentation fault (core dumped)
```

But if I don' t remove nodes, this code could work. 

Another problem is, if I replace the writing format with, for instance, nk.Format.LFR, this code could also work. But all the nodes and edges deleted before saving seems to appear again.

**Comment by Jerevia** [comment]
FYI，community detection algorithms, I tried louvain and leiden, give bad results or core dumped if removing nodes before running this algorithms

**Comment by Jerevia** [comment]
@Relux-the-Relux Hi, do you have any plans for fixing this issue?

**Comment by axelbrowne** [comment]
I just wrote added an issue before seeing that this is the same issue. Whoops. Here's my write-up on the same thing.

### Reading from NetworkitBinary file segfaults for graphs written with uncontinuous node-indices #996

graphio.writeGraph will write a graph with uncontinuous node indices (i.e. nodes are not indexed [0...G.numberOfNodes()) ) to a .networkit file without error, but reading from the same file will result in a segmentation fault.

Ex:
```
# G : a nk.Graph with continuous node indexing
# nodes : a list of uncontinuous/arbitrary nodes indices
# path : some file path string. doesn't matter


g = nk.graphtools.GraphTools.subgraphFromNodes(G, nodes)

# this passes
assert g.numberOfNodes() == len(nodes)

# runs successfully
nk.writeGraph(g, path, nk.Format.NetworkBinary)
```

Later...
```
>>> g = nk.readGraph(path, nk.Format.NetworkBinary)
Segmentation Fault (Core Dumped)
```

**Comment by fabratu** [comment]
If I am not mistaken, this issue is fixed with #1009 (bug in the writer) and therefore available in NetworKit version 10.1. To avoid this, you have to write the graph using 10.1.

Will close this issue for now. Feel free to reopen it if the problem still persists.
